Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -342,7 +342,7 @@ open class ChatChannelVC: _ViewController,
         willDisplayMessageAt indexPath: IndexPath
     ) {
         let message = chatMessageListVC(vc, messageAt: indexPath)
-        if message?.id == firstUnreadMessageId {
+        if message?.id == firstUnreadMessageId || firstUnreadMessageId == nil {
```

Comment:
firstUnreadMessageId is initially nil in most of the cases, because of the lifecycle, so this might not be enough

Reply:
This seems to be working fine 🤔

Reply:
Have you tried after the latest changes? (moving back to keeping the state on the UI)

Reply:
Yes it is working fine 🤔

Reply:
This should be fine, because whenever we mark the channel read, the `firstUnreadMessageId` will be `nil` and this is what we need to make sure the unread badge or banner does not appear for every new message